parlai/core/torch_agent.py

```diff
@@ -1969,10 +1974,11 @@ def state_dict(self):
         """
         states = {}
         if hasattr(self, 'model'):  # save model params
-            if hasattr(self.model, 'module'):
-                # did we wrap in a DistributedDataParallel
+            if hasattr(self.model, 'module') and not is_fsdp(self.model):
```

Comment:
nit: could make this a helper function too? like `should_sync_gradnorm` (not necessary of course)